### PR TITLE
Add rs.ruffle.Ruffle

### PR DESCRIPTION
**Pull request by Doomsdayrs**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
  - `--device=all` for controllers and webcam support.
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://github.com/ruffle-rs/ruffle/issues/9220
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*
  - appstream & desktop file upstreaming: https://github.com/ruffle-rs/ruffle/pull/11227
  - logo.svg upstreaming: https://github.com/ruffle-rs/ruffle/pull/11226

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: rs.ruffle.Ruffle.yaml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+sdk: org.freedesktop.Sdk
+#sdk-extensions:
+#  -  org.freedesktop.Sdk.Extension.rust-stable
+command: ruffle
+
+finish-args:
+#  - --socket=wayland # Disabled due to faulty libraries
+# <x11
+  - --socket=x11
+  - --socket=fallback-x11
```
Comment (hfiguiere) [inline rs.ruffle.Ruffle.yaml:13]:
```suggestion
  - --socket=x11
```

File: cargo-sources.json
Comment (barthalion) [inline cargo-sources.json:1]: This doesn't seem used.

File: rs.ruffle.Ruffle.metainfo.xml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+<!-- Copyright 2023 Doomsdayrs -->
+<component type="desktop">
+  <id>rs.ruffle.Ruffle</id>
+  <metadata_license>CC0-1.0</metadata_license>
+  <name>Ruffle</name>
+  <summary>A Flash Player emulator written in Rust</summary>
+  <launchable type="desktop-id">rs.ruffle.Ruffle.desktop</launchable>
+  <screenshots>
+    <screenshot type="default">
+      <image>https://raw.githubusercontent.com/flathub/flathub/rs.ruffle.Ruffle/assets/launcher.png</image>
```
Comment (barthalion) [inline rs.ruffle.Ruffle.metainfo.xml:11]: These should point to your fork until the merge.

File: rs.ruffle.Ruffle.metainfo.xml [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    <control>pointing</control>
+    <control>keyboard</control>
+    <control>gamepad</control>
+  </supports>
+  <project_license>MIT OR Apache-2.0</project_license>  
+  <developer_name>ruffle-rs</developer_name>
+  <content_rating type="oars-1.1">
+  </content_rating>
+  <update_contact>doomsdayrs@gmail.com</update_contact> <!-- Helped setup the flatpak -->
+  <update_contact>admin@ruffle.rs</update_contact> <!-- This is a placeholder -->
```
Comment (barthalion) [inline rs.ruffle.Ruffle.metainfo.xml:65]: I'm not actually sure you can define it twice, and it seems misleading given no involvement from upstream.

File: rs.ruffle.Ruffle.yaml [hunk 4]
```diff
@@ -0,0 +1,65 @@
+app-id: rs.ruffle.Ruffle
+runtime: org.freedesktop.Platform
+runtime-version: '22.08'
+sdk: org.freedesktop.Sdk
+#sdk-extensions:
+#  -  org.freedesktop.Sdk.Extension.rust-stable
```
Comment (barthalion) [inline rs.ruffle.Ruffle.yaml:6]: Not used.

File: rs.ruffle.Ruffle.yaml [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+  - --share=ipc
+# x11>
+  - --socket=pulseaudio
+  - --share=network # Flash can connect to the internet
+  - --device=all # controllers (?)
+
+#build-options:
+#  append-path: /usr/lib/sdk/rust-stable/bin
+#  env:
+#    CARGO_HOME: /run/build/ruffle/cargo
```
Comment (barthalion) [inline rs.ruffle.Ruffle.yaml:22]: Not used.

File: rs.ruffle.Ruffle.yaml [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+modules:
+  - name: rust
```
Comment (barthalion) [inline rs.ruffle.Ruffle.yaml:25]:
```suggestion
  - name: ruffle
```

File: rs.ruffle.Ruffle.yaml [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+#    build-commands:
+#      - pwd
+#      - ls $CARGO_HOME
+#      - cargo --offline fetch --manifest-path Cargo.toml
+#      - cargo --offline build --release --package=ruffle_desktop
+#    sources:
+#      - cargo-sources.json
+#      - type: git
+#        url: https://github.com/ruffle-rs/ruffle.git
+#        tag: nightly-2023-01-19
```
Comment (barthalion) [inline rs.ruffle.Ruffle.yaml:65]: Not used.

File: rs.ruffle.Ruffle.yaml [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+runtime-version: '22.08'
+sdk: org.freedesktop.Sdk
+#sdk-extensions:
+#  -  org.freedesktop.Sdk.Extension.rust-stable
+command: ruffle
+
+finish-args:
+  - --socket=wayland
+# <x11
+  - --socket=x11 # required for arboard fault
```
Comment (barthalion) [inline rs.ruffle.Ruffle.yaml:12]: I assume you meant `fallback-x11`  here?

Comment (Doomsdayrs) [reply rs.ruffle.Ruffle.yaml:12]: Nope! Ruffle has a hard dependency on x11 due to a currently unpatchable unsolvable fault in arboard

File: cargo-sources.json
Comment (Doomsdayrs) [reply cargo-sources.json:1]: This was for building from source, which was not working due to issues with network requirement during build.

Comment (Doomsdayrs) [reply cargo-sources.json:1]: I will remove it.

File: rs.ruffle.Ruffle.yaml (hunk 8 above)
Comment (barthalion) [reply rs.ruffle.Ruffle.yaml:10]: If it works strictly only on X11, there's no point in allowing Wayland socket.

Comment (Doomsdayrs) [reply rs.ruffle.Ruffle.yaml:10]: *but*, it still renders with wayland. The backend library just needs to spawn an x11 window to momentarily work.

**Comment by barthalion** [comment]
bot, build rs.ruffle.Ruffle

**Comment by barthalion** [comment]
bot, build rs.ruffle.Ruffle

**Comment by JakobDev** [comment]
What's the state of this?

**Comment by Doomsdayrs** [comment]
> What's the state of this?

Just this https://github.com/ruffle-rs/ruffle/pull/11227 . After it is merged this launcher won't have to make its own meta info.

**Comment by hfiguiere** [comment]
bot, build rs.ruffle.Ruffle

**Comment by Doomsdayrs** [comment]
Alright, Upstream has some hesitation with the appstream. We can continue to merge this and I shall accommodate in the future.

File: README.md [hunk 9]
```diff
@@ -0,0 +1,14 @@
+# [Ruffle](https://ruffle.rs/)
+
+ruffle is a Flash Player emulator built in the Rust programming language.
+
+## Issues
+
+- Currently this does not build from source, it just downloads the x86 binary. The naga package fails due to wgpu.
+- Hard depencnecy on x11.
```
Comment (hfiguiere) [inline README.md:8]:
```suggestion
- Hard dependency on x11.
```

File: rs.ruffle.Ruffle.metainfo.xml [hunk 10]
```diff
@@ -0,0 +1,73 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Copyright 2023 Doomsdayrs -->
+<component type="desktop">
+  <id>rs.ruffle.Ruffle</id>
+  <metadata_license>CC0-1.0</metadata_license>
+  <name>Ruffle</name>
+  <summary>A Flash Player emulator written in Rust</summary>
```
Comment (hfiguiere) [inline rs.ruffle.Ruffle.metainfo.xml:7]:
```suggestion
  <summary>A Flash Player emulator</summary>
```

whatever is used to write is is irrelevant to users.

File: rs.ruffle.Ruffle.desktop [hunk 11]
```diff
@@ -0,0 +1,10 @@
+[Desktop Entry]
+Type=Application
+Name=Ruffle
+GenericName=Flash Player
+Comment=A Flash Player emulator written in Rust
```
Comment (hfiguiere) [inline rs.ruffle.Ruffle.desktop:5]:
```suggestion
Comment=A Flash Player emulator
```

**Comment by Doomsdayrs** [comment]
Sorry for the delay, It seems the email got buried.

**Comment by Doomsdayrs** [comment]
bot, build rs.ruffle.Ruffle

**Comment by Doomsdayrs** [comment]
As with https://github.com/ruffle-rs/ruffle/pull/14362, the clipboard issue is resolved. Tomorrow I will send out a commit with tomorrows (working) version of Ruffle without x11 flags.

**Comment by puigru** [comment]
bot, build rs.ruffle.Ruffle

File: flathub.json [hunk 12]
```diff
@@ -0,0 +1,4 @@
+{
+	"only-arches": ["x86_64"],
+	"automerge-flathubbot-prs": true
```
Comment (bbhtt) [inline flathub.json:3]: automerge should be removed

File: assets/btd.png
Comment (bbhtt) [inline assets/btd.png:1]: These screenshots should be upstream

File: rs.ruffle.Ruffle.desktop
Comment (bbhtt) [inline rs.ruffle.Ruffle.desktop:1]: Desktop file should be upstream

File: rs.ruffle.Ruffle.yaml [hunk 13]
```diff
@@ -0,0 +1,41 @@
+app-id: rs.ruffle.Ruffle
+runtime: org.freedesktop.Platform
+runtime-version: '22.08'
```
Comment (bbhtt) [inline rs.ruffle.Ruffle.yaml:3]:
```suggestion
runtime-version: '23.08'
```

File: ruffle
Comment (bbhtt) [inline ruffle:1]: Why is this a submodule?

**Comment by bbhtt** [comment]
bot, build rs.ruffle.Ruffle

**Comment by bbhtt** [comment]
Please switch to upstream metadata once https://github.com/ruffle-rs/ruffle/pull/11227 is merged.

**Comment by evilpie** [comment]
The metadata just landed upstream in ruffle.

**Comment by kjarosh** [comment]
Created a new PR based on this one with all the issues fixed (hopefully):

* https://github.com/flathub/flathub/pull/5473

Thank you again for your work @Doomsdayrs !

**Comment by bbhtt** [comment]
Closing this in favour of https://github.com/flathub/flathub/pull/5473

**Pull request closed without merging**